lib/modules/platform/codecommit/index.ts

```diff
@@ -0,0 +1,714 @@
+import JSON5 from 'json5';
+import { TextDecoder } from 'web-encoding';
+import {
+  PLATFORM_BAD_CREDENTIALS,
+  REPOSITORY_EMPTY,
+  REPOSITORY_NOT_FOUND,
+} from '../../../constants/error-messages';
+import { logger } from '../../../logger';
+import { BranchStatus, PrState, VulnerabilityAlert } from '../../../types';
+import * as git from '../../../util/git';
+import { regEx } from '../../../util/regex';
+import { sanitize } from '../../../util/sanitize';
+import type {
+  BranchStatusConfig,
+  CreatePRConfig,
+  EnsureCommentConfig,
+  EnsureCommentRemovalConfig,
+  EnsureIssueConfig,
+  EnsureIssueResult,
+  FindPRConfig,
+  Issue,
+  MergePRConfig,
+  PlatformParams,
+  PlatformResult,
+  Pr,
+  RepoParams,
+  RepoResult,
+  UpdatePrConfig,
+} from '../types';
+import { getNewBranchName, repoFingerprint } from '../util';
+import { smartTruncate } from '../utils/pr-body';
+import * as client from './codecommit-client';
+import { getUserArn, initIamClient } from './iam-client';
+import { getCodeCommitUrl } from './util';
+
+const decoder = new TextDecoder();
+
+interface Config {
+  repository?: string;
+  defaultBranch?: string;
+  region?: string;
+  prList?: Pr[];
+  credentials?: Credentials;
+  userArn?: string;
+}
+
+const config: Config = {};
+
+export async function initPlatform({
+  endpoint,
+  username,
+  password,
+}: PlatformParams): Promise<PlatformResult> {
+  let accessKeyId = username;
+  let secretAccessKey = password;
+  let region;
+
+  if (!accessKeyId) {
+    accessKeyId = process.env.AWS_ACCESS_KEY_ID;
+  }
+  if (!secretAccessKey) {
+    secretAccessKey = process.env.AWS_SECRET_ACCESS_KEY;
+  }
+  if (endpoint) {
+    const regionReg = regEx(/.*codecommit\.(?<region>.+)\.amazonaws\.com/);
+    const codeCommitMatch = regionReg.exec(endpoint);
+    region = codeCommitMatch?.groups?.region;
+    if (!region) {
+      logger.warn("Can't parse region, make sure your endpoint is correct");
+    }
+  } else {
+    region = process.env.AWS_REGION;
+  }
+
+  if (!accessKeyId || !secretAccessKey || !region) {
+    throw new Error(
+      'Init: You must configure a AWS user(accessKeyId), password(secretAccessKey) and endpoint/AWS_REGION'
+    );
+  }
+
+  config.region = region;
+  const credentials = {
+    accessKeyId,
+    secretAccessKey,
+    sessionToken: process.env.AWS_SESSION_TOKEN,
+  };
+  config.credentials = credentials;
+
+  initIamClient(region, credentials);
+  config.userArn = await getUserArn();
+
+  client.buildCodeCommitClient(region, credentials);
+
+  const platformConfig: PlatformResult = {
+    endpoint: endpoint ?? `https://git-codecommit.${region}.amazonaws.com/`,
+  };
+  return Promise.resolve(platformConfig);
+}
+
+export async function initRepo({
+  repository,
+  endpoint,
+}: RepoParams): Promise<RepoResult> {
+  logger.debug(`initRepo("${repository}")`);
+
+  config.repository = repository;
+  const url = getCodeCommitUrl(config.region!, repository, config.credentials!);
```

Comment:
are region and cred are definitely defined? what happens if they are not?
wouldn't it be better to if check them before function invocation?

Reply:
i do in the initPlatform